src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -177,10 +177,11 @@ def check_user
       User.session = user_checker.find_or_create_user!
 
       if User.session!.is_active?
-        User.session!.update_user_info_from_proxy_env(request.env)
+        User.session!.update_login_values(request.env)
       else
         session[:login] = nil
         User.session = User.find_nobody!
+        User.session!.count_login_failure
```

Comment:
Hm, I don't get it. `User.session!` will just always raise here right?

Reply:
First, I've moved `User.session!.count_login_failure` two lines up because we need `User.session` to be the real user and not `nobody`. If I'm not wrong, at this point the real user is there but its state is not `confirmed`/not active. Shouldn't we consider it as an error?